client/src/fluid/Fluid.ml

```diff
@@ -1902,8 +1911,36 @@ let replacePartialWithArguments
                in
                let oldParams = getFunctionParams oldName count existingExprs in
                let matchedParams, mismatchedParams =
-                 List.partition oldParams ~f:(fun p ->
-                     List.any newParams ~f:(isAligned p))
+                 List.partitionMap oldParams ~f:(fun p ->
+                     let comparedResults =
+                       List.map newParams ~f:(compareParams p)
+                     in
+                     let alignedIndex =
+                       List.findIndex comparedResults ~f:(fun r ->
+                           r = CPAligned)
+                     in
+                     let typeAndPositionMatchedIndex =
+                       List.findIndex comparedResults ~f:(fun r ->
+                           r = CPTypeAndPositionMatched)
+                     in
+                     let matchedNewParam =
+                       match (alignedIndex, typeAndPositionMatchedIndex) with
+                       | Some index, _ | None, Some index ->
+                           let np = List.getAt ~index newParams in
+                           ( match (np, p) with
+                           | ( Some (Some (name, tipe, _, index))
+                             , Some (_, _, expr, _) ) ->
+                               Some (name, tipe, expr, index)
+                           | _ ->
+                               None )
+                       | None, None ->
+                           None
+                     in
+                     match matchedNewParam with
+                     | Some (name, tipe, expr, index) ->
+                         Left (Some (name, tipe, expr, index))
```

Comment:
I think this could be folded into the previous match statement.

Reply:
Good point, changed.